### PR TITLE
chore(datastore): Add samples from ruby-docs-samples

### PR DESCRIPTION
**Pull request by quartzmo**

### CODE REVIEW

File: google-cloud-datastore/samples/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Cloud Datastore Sample
+
+This sample provides example code for
+[cloud.google.com/datastore/docs](https://cloud.google.com/datastore/docs).
+
+## Setup
+
+Before you can run or test the sample, you will need to enable the Cloud Datastore API in the [Google Developers Console](https://console.developers.google.com/projectselector/apis/api/datastore/overview).
```
Comment (dazuma) [inline google-cloud-datastore/samples/README.md:8]: `[Google Cloud Console](https://console.cloud.google.com/apis/library/datastore.googleapis.com)`

File: google-cloud-datastore/samples/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* `GCLOUD_KEYFILE_JSON`
+* `GOOGLE_CLOUD_KEYFILE_JSON`
+
+For more information, see
+[Authentication](https://googlecloudplatform.github.io/gcloud-ruby/#/docs/guides/authentication).
+
+### Creating the Datastore indexes
+
+Install the [gcloud command-line
+tool](https://developers.google.com/cloud/sdk/gcloud/) and use it to create the
```
Comment (dazuma) [inline google-cloud-datastore/samples/README.md:41]: `https://cloud.google.com/sdk/gcloud`

File: google-cloud-datastore/samples/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+* `DATASTORE_KEYFILE`
+* `GCLOUD_KEYFILE`
+* `GOOGLE_CLOUD_KEYFILE`
+* `DATASTORE_KEYFILE_JSON`
+* `GCLOUD_KEYFILE_JSON`
+* `GOOGLE_CLOUD_KEYFILE_JSON`
+
+For more information, see
+[Authentication](https://googlecloudplatform.github.io/gcloud-ruby/#/docs/guides/authentication).
```
Comment (dazuma) [inline google-cloud-datastore/samples/README.md:36]: `https://googleapis.dev/ruby/google-cloud-datastore/latest/file.AUTHENTICATION.html`

File: google-cloud-datastore/samples/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+service and require authentication.
+
+### Authenticating
+
+Set one of the following environment variables to your Google Cloud Platform
+project ID:
+
+* `DATASTORE_DATASET`
+* `DATASTORE_PROJECT`
+* `GCLOUD_PROJECT`
```
Comment (dazuma) [inline google-cloud-datastore/samples/README.md:22]: Don't list the `GCLOUD_*` environment variables, as we consider those deprecated.

File: google-cloud-datastore/samples/acceptance/quickstart_test.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+require_relative "../quickstart.rb"
+
+describe "Datastore Quickstart" do
+  let(:datastore) { Google::Cloud::Datastore.new }
+  let(:task_key) { datastore.key "Task", "sampletask1" }
+
+  before do
+    if (task = datastore.find task_key)
+      datastore.delete task
+    end
```
Comment (dazuma) [inline google-cloud-datastore/samples/acceptance/quickstart_test.rb:25]: Probably less awkward for something like this:
```ruby
task = datastore.find task_key
datastore.delete task if task
```

File: google-cloud-datastore/samples/acceptance/tasks_test.rb [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  let(:datastore) { Google::Cloud::Datastore.new }
+
+  before do
+    delete_tasks
+  end
+
+  def wait_until times: 5
+    times.times do
+      return if yield
+    end
```
Comment (dazuma) [inline google-cloud-datastore/samples/acceptance/tasks_test.rb:28]: Maybe there should be a sleep between attempts? Are the uses of this stable enough?

Comment (quartzmo) [reply google-cloud-datastore/samples/acceptance/tasks_test.rb:28]: I agree, looks like an error in the original method. I'll add a 1 s sleep.

Comment (quartzmo) [reply google-cloud-datastore/samples/acceptance/tasks_test.rb:29]: Done.

File: google-cloud-datastore/samples/acceptance/quickstart_test.rb (hunk 5 above)
Comment (quartzmo) [reply google-cloud-datastore/samples/acceptance/quickstart_test.rb]: Done.

File: google-cloud-datastore/samples/README.md (hunk 4 above)
Comment (quartzmo) [reply google-cloud-datastore/samples/README.md]: Done.

File: google-cloud-datastore/samples/README.md (hunk 3 above)
Comment (quartzmo) [reply google-cloud-datastore/samples/README.md]: Done.

File: google-cloud-datastore/samples/README.md (hunk 2 above)
Comment (quartzmo) [reply google-cloud-datastore/samples/README.md]: Done.

File: google-cloud-datastore/samples/README.md (hunk 1 above)
Comment (quartzmo) [reply google-cloud-datastore/samples/README.md]: Done.

**Comment by quartzmo** [comment]
CI is failing because the indexes haven't been created:

```
  1) Error:
Datastore sample#test_0017_supports basic_query run_query:
Google::Cloud::FailedPreconditionError: 9:no matching index found. recommended index is:
- kind: Task
  properties:
  - name: done
  - name: priority
    direction: desc
```

**Pull request merged**